### ISSUE TITLE
Prioritize "Use Type Signature" action when available

### ISSUE THREAD

**Issue by ChrisPenner** [haskell/haskell-language-server#457]
Hey folks! Amazing job on this project! Really appreciate how wonderful this tool is 😄 

The LSP client I'm using offers a "quick-fix" binding; which AFAICT just executes the first "code action" related to your cursor.

The issue I'm having relates to type holes; 

E.g. If I write the binding:

```haskell
thing :: _
thing = 'a'
```

If I have my cursor on the type hole and run "quick-fix" in the Language server it executes the FIRST code action, which in my case seems to always be `Add PartialTypeSignatures`; when what I (and probably most folks) **always** want is the `Use type signature: '(Char)'` option (which shows up second in the list if I list them all).

I'm not sure if it's possible to control the ordering of code-action suggestions, but if it is It'd be great if "Use type signature" would always come first if available 😄 

### Your environment
* Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`
```
haskell-language-server version: 0.4.0.0 (GHC: 8.10.1) (PATH: /Users/cpenner/.ghcup/bin/haskell-language-server-wrapper-0.4.0) (GIT hash: 0a18edde24923251a148cbbc0ae993a6aac83b9c)
Tool versions found on the $PATH
cabal:          3.2.0.0
stack:          2.3.1
ghc:            8.8.4
```
## Which lsp-client do you use
  * Neovim - COC.vim

**Comment by tek** [comment]
I wonder if this is generally a thing that should be configurable in the language client, with string matching. I find that I don't want the first action quite frequently.

**Comment by jneira** [comment]
>I wonder if this is generally a thing that should be configurable in the language client, with string matching

Agree, not sure if we could establish a sensible for everybody ordering in the server. Configuring seems to be complex too.

**Comment by konn** [comment]
Perhaps providing a option to "place most recent choice at the top" could make sense? (Then we need a way to maintain such history somewhere and it seems rather nontrivial, though...)

**Comment by jneira** [comment]
> Perhaps providing a option to "place most recent choice at the top" could make sense?

That sounds great, and the most used one in recent editors. Anyways it should be implemented in the client for sure.
